Sources/Scheduler.swift

```diff
@@ -193,8 +193,14 @@ public final class QueueScheduler: DateScheduler {
 
 	public let queue: DispatchQueue
 
+	private let lock: Lock
+	
+	private var timers: [DispatchSourceTimer]
```

Comment:
You may use `Atomic`. I'd prefer `Set` which fits the scenario here (insert/removal only).

Reply:
`DispatchSourceTimer` isn't `Hashable`, so I think it would have to be an `NSMutableSet`. Is that an improvement over using a Swift array?